### ISSUE TITLE
Backfill revisit

### ISSUE THREAD

**Issue by chenziliang** [timeplus-io/proton#162]
**Use case**

In Proton, data retention policy is different in streaming store and historical data store. When we rewind data (`_tp_time > now - 7d` for example), there is possibility that the data is already gone in streaming store. In such case, we will need backfill the data from the historical data store and when the backfill is done, re-connects it with the live stream events.

Backfilling data from the historical data store may require `re-sort` the data depending on the query semantics. For example, in streaming tumble / hop / session etc window aggregation, we will need resort the data according to `_tp_time` or the logic time sequence number `_tp_sn`. 

`_tp_sn` is not indexed in the historical data store by default but can enable it automatically when a stream has `_tp_sn` column during provision. Sorting by `_tp_sn` could be very expensive since data in the historical data store is usually not sorted by sequence number. 

`_tp_time` is indexed in the historical data store by default for every event in every stream and data is sorted by `_tp_time` in coarse granularity there. So re-sort data by `_tp_time` is usually way cheaper than `_tp_sn`.

From most application business point of view, when the data is already past / historical, backfill in `_tp_time` (event time) order seems making lots of sense instead of in the logic `_tp_sn` sequence number order (ingest time order).

So this issue tries to make backfill smarter and reasonable for query like below. Once this enhancement is done, it will automatically backfill events from the historical data store with necessary `sorting`.

```
SELEC max(cpu), min(cpu), device 
FROM tumble(device_utils, 5m)
WHERE _tp_time > now() - 7d;
```

For query like below, we don't need resort historical data during backfill since global aggregation doesn't care time / temporal. 
```
SELEC max(cpu), min(cpu), device 
FROM device_utils
WHERE _tp_time > now() - 7d;
```

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by chenziliang** [comment]
If we don't need emit intermediate aggr result, we probably don't need sort the data. We can introduce a setting to control this behavior. For example,

```
SELEC max(cpu), min(cpu), device 
FROM device_utils
WHERE _tp_time > now() - 7d;
SETTINGS emit_during_backfill = false;
```

**Comment by yl-lisen** [comment]
> If we don't need emit intermediate aggr result, we probably don't need sort the data. We can introduce a setting to control this behavior. For example,
> 
> ```
> SELEC max(cpu), min(cpu), device 
> FROM device_utils
> WHERE _tp_time > now() - 7d;
> SETTINGS emit_during_backfill = false;
> ```

I think it's an optimized behavior of global aggregation (not apply for window aggregation)
For global aggregation, by default, we should skip emit during backfill. The setting `emit_during_backfill ` not required

**Comment by chenziliang** [comment]
end user may still like to see the intermediate results for global aggr.

**Comment by jovezhong** [comment]
(Jove Github Bot) assuming it is not done, deferred this ticket to the next sprint.
